ariadne/cardano/src/Ariadne/Wallet/Cardano/Kernel/Keystore.hs

```diff
@@ -117,23 +87,23 @@ bracketKeystore deletePolicy fp withKeystore =
 
 -- | Creates a new keystore.
 newKeystore :: FilePath -> IO Keystore
-newKeystore fp = runIdentityT $ fromKeystore $ do
-    us <- peekUserSecret fp
-    Keystore <$> newMVar (InternalStorage us)
+newKeystore fp = do
+    istorage <- peekInternalStorage fp
+    Keystore <$> newMVar istorage
 
--- | Creates a legacy 'Keystore' by reading the 'UserSecret' from a 'NodeContext'.
+-- | Creates a legacy 'Keystore' by reading the 'InternalStorage' from a 'NodeContext'.
 -- Hopefully this function will go in the near future.
-newLegacyKeystore :: UserSecret -> IO Keystore
-newLegacyKeystore us = Keystore <$> newMVar (InternalStorage us)
+newLegacyKeystore :: InternalStorage -> IO Keystore
+newLegacyKeystore istorage = Keystore <$> newMVar istorage
 
 -- | Creates a legacy 'Keystore' using a 'bracket' pattern, where the
 -- initalisation and teardown of the resource are wrapped in 'bracket'.
 -- For a legacy 'Keystore' users do not get to specify a 'DeletePolicy', as
 -- the release of the keystore is left for the node and the legacy code
 -- themselves.
-bracketLegacyKeystore :: UserSecret -> (Keystore -> IO a) -> IO a
-bracketLegacyKeystore us withKeystore =
-    bracket (newLegacyKeystore us)
+bracketLegacyKeystore :: InternalStorage -> (Keystore -> IO a) -> IO a
```

Comment:
Let's get rid of legacy. That's the primary goal of the issue. Fortunately, this function seems to be unused, so it can be simply deleted.